inc/class-thumbnails.php

```diff
@@ -92,7 +93,16 @@ public function check_orphaned( $post_id, $post ) {
 				if ( strpos( $meta_key, '_' . $provider . '_' ) !== false && strpos( $meta_key, '_url' ) === false ) {
 					$id = str_replace( 'embed_privacy_thumbnail_' . $provider . '_', '', $meta_key );
 					
-					if ( strpos( $post->post_content, $id ) === false ) {
+					$missing_id = strpos( $post->post_content, $id ) === false;
+					$missing_url = true;
+					if ( $missing_id && isset( $metadata[ $meta_key . '_url' ] ) ) {
+						$url = $metadata[ $meta_key . '_url' ];
+						if ( is_array( $url ) ) {
```

Comment:
Why can this be an array? There should always be only one URL (the one entered in the editor) stored in the database.

Reply:
I have no idea, but the same check was on line 88 and 144:

https://github.com/Lazza/embed-privacy/blob/28aa9faa2d643201099c7627430d458f8935d10c/inc/class-thumbnails.php#L88

https://github.com/Lazza/embed-privacy/blob/28aa9faa2d643201099c7627430d458f8935d10c/inc/class-thumbnails.php#L144

I simply copied that one.

Also, in my tests it was an array. It makes no sense but I am not so expert in the WP codebase to ascertain why it is the case.

Reply:
Oh yeah, got it. A post can contain multiple links to multiple embeds and thus also to multiple thumbnails.